Documentation/network/clustermesh/clustermesh.rst

```diff
@@ -65,6 +65,32 @@ Additional Requirements for Native-routed Datapath Modes
   firewall rules allowing pods in different clusters to reach each other on all
   ports.
 
+Cluster Identity Requirements
+=============================
+
+* All clusters must be configured with the same cluster identity mode using the
+  ``maxConnectedClusters`` option. By default, up to 255 clusters can be connected together. This
+  number can be increased to 511 by supplying ``maxConnectedClusters``:
+
+ * ConfigMap option ``max-connected-clusters=511``
+ * Helm option ``--set maxConnectedClusters=511``
+ * ``cilium install`` option ``--set maxConnectedClusters=511``
+
+* This option controls the bit allocation of numeric identities and will affect the number of
+  identities that can be allocated per cluster:
+
++------------------------+------------+----------+----------+
+| MaxConnectedClusters   | Maximum cluster-local identities |
++========================+============+==========+==========+
+| 255                    | 65535                            |
++------------------------+------------+----------+----------+
+| 511                    | 32767                            |
++------------------------+------------+----------+----------+
+
+.. warning::
+  ``MaxConnectedClusters`` can only be set once during Cilium installation and should not be
+  changed for existing clusters.
```

Comment:
I'd suggest extending a bit on the consequences of changing the `MaxConnectedClusters` value, something like connection disruption and possible incorrect enforcement of network policies.